attachment.php

```diff
@@ -83,6 +83,15 @@
 	{
 		error($lang->error_invalidattachment);
 	}
+
+	$attachtypes = (array)$cache->read('attachtypes');
+	$ext = get_extension($attachment['filename']);
+	$attachtype = $attachtypes[$ext];
```

Comment:
This + lines below may produce PHP notices, some kind of `isset()`/`!empty()` check would be good.